cmd/algons/dnsaddrCmd.go

```diff
@@ -63,3 +80,93 @@ var dnsaddrTreeCmd = &cobra.Command{
 		}
 	},
 }
+var dnsaddrTreeDeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Recursively resolves and deletes the dnsaddr entries of the given domain",
+	Long:  "Recursively resolves and deletes the dnsaddr entries of the given domain",
+	Run: func(cmd *cobra.Command, args []string) {
+		addr, err := multiaddr.NewMultiaddr(fmt.Sprintf("/dnsaddr/%s", dnsaddrDomain))
+		if err != nil {
+			fmt.Printf("unable to construct multiaddr for %s : %v\n", dnsaddrDomain, err)
+			return
+		}
+		controller := dnsaddr.NewMultiaddrDNSResolveController(secure, "")
+		cfZoneID, cfToken, err := getClouldflareCredentials()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error getting DNS credentials: %v", err)
+			return
+		}
+		cloudflareDNS := cloudflare.NewDNS(cfZoneID, cfToken)
+		var recordsToDelete []cloudflare.DNSRecordResponseEntry
+		err = dnsaddr.Iterate(addr, controller, func(entryFrom multiaddr.Multiaddr, entries []multiaddr.Multiaddr) error {
+			domain, _ := entryFrom.ValueForProtocol(multiaddr.P_DNSADDR)
+			name := fmt.Sprintf("_dnsaddr.%s", domain)
+			fmt.Printf("listing records for %s\n", name)
+			records, err0 := cloudflareDNS.ListDNSRecord(context.Background(), "TXT", name, "", "", "", "")
+			if err0 != nil {
+				fmt.Printf("erroring listing dns records for %s %s\n", domain, err)
+				return err
+			}
+			for _, record := range records {
+				fmt.Printf("found record to delete %s:%s\n", record.Name, record.Content)
+				recordsToDelete = append(recordsToDelete, record)
+			}
+			return nil
+		})
+		if err != nil {
+			fmt.Printf("%s\n", err.Error())
+			return
+		}
+		err = checkedDelete(recordsToDelete, cloudflareDNS)
+		if err != nil {
+			fmt.Printf("error deleting records: %s\n", err)
+		}
+	},
+}
+
+var dnsaddrTreeCreateCmd = &cobra.Command{
+	Use:   "create",
```

Comment:
Should you try listing first to make sure nothing exists at this path? Once we start using this for production, I think it would be easy to mistake `create` with the old `add` approach. Maybe an additional `add` command that deletes everything to make room for the new tree would be helpful as well.

Reply:
[95efdb7b73d8804e8a62fc0f4fb5645a0f123973](https://github.com/algorand/go-algorand/pull/5631/commits/95efdb7b73d8804e8a62fc0f4fb5645a0f123973)